src/XMakeBuildEngine/Evaluation/Expander.cs

```diff
@@ -170,6 +175,12 @@ internal Expander(IPropertyProvider<P> properties, IItemProvider<I> items, IMeta
             _metadata = metadata;
         }
 
+        public Expander(IPropertyProvider<P> properties, IItemProvider<I> items, IItemFactory<I, I> defaultItemFactory) : this(properties, items)
```

Comment:
I usually make the type internal and leave the members public.

I am still fleshing out the interaction with the expander. I just found out that I will actually have to mutate the default item factory from each item spec, which is a bit of a terrible thing, so most likely I'll end up undoing the expander changes and have the item spec receive an IValuatorData instead of an expander. Using that, an itemspec could create it's own personal expander that would not risk destroying the state of other users of the same expander object.